### PR TITLE
Plugins and protocols manifest files are now required

### PR DESCRIPTION
**Pull request by scottinet**
# Description

In Kuzzle v1, it was possible to create plugins or protocols without a manifest.json file.
In that case, Kuzzle logged a warning explaining that it was assuming that the plugin/protocol was compatible with Kuzzle v1, and then fallbacked to the package.json file.
The warning also stated that this fallback will cease to exist in v2.

So. Kuzzle v2 will be released, and this obsoletes this fallback.
That PR removes it, making Kuzzle crash with a meaningful error if a plugin or a protocol does not feature a manifest.json file.

### CODE REVIEW

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [pluginsManager.js#L968](https://github.com/kuzzleio/kuzzle/blob/89f9c0e7fa70428f0d27510eb7cda328987ddd6f/lib/api/core/plugins/pluginsManager.js#L968): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

File: lib/api/core/entrypoints/manifest.js [hunk 1]
```diff
@@ -19,9 +19,7 @@
  * limitations under the License.
```
Comment (Aschen) [inline lib/api/core/entrypoints/manifest.js]: Shouldn't we get ride of this file ? I don't see the point since it's almost empty

Comment (scottinet) [reply lib/api/core/entrypoints/manifest.js]: It inherits AbstractManifest, which is also implemented by https://github.com/kuzzleio/kuzzle/blob/manifest-files-are-required/lib/api/core/plugins/manifest.js

I think it's best to keep this implementation that way: there are always slight differences between protocols and plugins manifest, and something tells me that the differences will only increase in the future.

Comment (Aschen) [reply lib/api/core/entrypoints/manifest.js]: Ok I see! Maybe we can move this file to a more explicit directory ? (Maybe we need a `protocols` directory?)

Comment (scottinet) [reply lib/api/core/entrypoints/manifest.js]: Sounds like a good idea.

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [pluginsManager.js#L968](https://github.com/kuzzleio/kuzzle/blob/952f0f584c07c7cd5019fea5aa9730daf1cc67f2/lib/api/core/plugins/pluginsManager.js#L968): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [pluginsManager.js#L968](https://github.com/kuzzleio/kuzzle/blob/6dbba22e42c17249a7c6776758850bbf27f92ffd/lib/api/core/plugins/pluginsManager.js#L968): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [pluginsManager.js#L968](https://github.com/kuzzleio/kuzzle/blob/d40d332a89d83f6f04244a8369f6ac948f4c473e/lib/api/core/plugins/pluginsManager.js#L968): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [pluginsManager.js#L968](https://github.com/kuzzleio/kuzzle/blob/58132d4af0d847117906da1cf954d3f4fd7f4b9b/lib/api/core/plugins/pluginsManager.js#L968): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Pull request merged**